services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -133,20 +134,50 @@ def scores
   end
 
   def my_account
-    @time_zones = [{name: 'Select Time Zone', id: 'Select Time Zone'}].concat(TZInfo::Timezone.all_country_zone_identifiers.sort.map{|tz| {name: tz.gsub('_', ' '), id: tz}})
+    session[:google_redirect] = request.env['PATH_INFO']
+    session[:clever_redirect] = request.env['PATH_INFO']
+    @google_or_clever_just_set = session[:google_or_clever_just_set]
+    session[:google_or_clever_just_set] = nil
+    @account_info = get_account_info
   end
 
-  def my_account_data
-    render json: current_user.generate_teacher_account_info
+  def update_my_account
+    response = current_user.update_teacher(params['classroom_manager'])
+    if response && response[:errors] && response[:errors].any?
+      errors = {}
+      if response[:errors]['email']&.include?('is being updated to a email that exists')
+        errors['email'] = ['That email is taken. Try another.']
+      elsif response[:errors]['email']&.include?('does not appear to be a valid e-mail address')
+        errors['email'] = ['Enter a valid email']
+      elsif response[:errors]['name']&.include?('must include first and last name')
+        errors['name'] = ["Enter both a first and last name"]
+      end
+      render json: {errors: errors}, status: 422
+    else
+      render json: get_account_info
+    end
   end
 
-  def update_my_account
-    response = current_user.update_teacher params
-    render json: response
+  def update_my_password
+    errors = {}
+    if current_user.authenticate(params[:current_password])
```

Comment:
This could be moved to the model in an `update_password` method that uses validations and returns the user record with errors if it's not successful.

Reply:
same as above, if that's okay.

Reply:
:+1: